### DISCUSSION TITLE
How to colorize the tailwind option on the auto complete popup

### DISCUSSION THREAD

**Discussion by itsbetma** [NvChad/NvChad#2972] [Q&A] [answered]
Hi, I am using the `tailwindcss-language-server` and the options suggestion is working well, but I am here to ask if it is possible to colorize the color options with it respective color?

![image](https://github.com/user-attachments/assets/07ae67d0-cc7f-4740-b8e0-f121042410cc)

Example of VSCode:
![image](https://github.com/user-attachments/assets/5d54bb3f-637a-4a86-a900-ff82ee1e6b01)

**Comment by siduck** [comment]
you     need to use a plugin for this like nvim-highlight, i'll however look for an easier way and tell u tomorrow!

**Reply by itsbetma** [reply]
Thanks, I will check into nvim-highlight and also wait for your answer.

**Reply by siduck** [reply, marked as answer]
@itsbetma  i managed to do it! 

![image](https://github.com/user-attachments/assets/1d0f20b0-23b8-4c29-a2c4-89e5828a961c)

and thats the format function 

![image](https://github.com/user-attachments/assets/09d42844-5369-46b9-8563-16ec10752060)

i'll add the support for this in nvchad itself soon! within this month :D

i couldnt believe it could be this easy, because tailwind lsp itself provides hex color for each color related completion so this feature  can be done with few lines of code + we dont have to hardcode tailwind colors in a file and even user custom colors in tailwind config work

**Reply by Rick-Phoenix** [reply]
@siduck how do I enable this in nvchad? 
I copied these configs from the wiki at :h nvui.cpm
```
M.ui = {
  cmp = {
    icons_left = true, 
    style = "default", 
    abbr_maxwidth = 60,
    format_colors = {
      tailwind = true, 
      icon = "󱓻",
    },
},
```
But I am not getting the color previews in cpm. This is the plugin config:

```
  {
    "hrsh7th/nvim-cmp",
    opts = function()
      local cmp = require "cmp"
      local calculated_opts = {
        sources = { 
          { name = "nvim_lsp" },
          { name = "luasnip" },
          { name = "buffer" },
          { name = "nvim_lua" },
          { name = "path" },
        },
        mapping = {
          ["<Up>"] = cmp.mapping(function(fallback)
            if cmp.visible() then
              cmp.select_prev_item()
            elseif require("luasnip").jumpable(-1) then
              require("luasnip").jump(-1)
            else
              fallback()
            end
          end, { "i", "s" }),
          ["<Down>"] = cmp.mapping(function(fallback)
            if cmp.visible() then
              cmp.select_next_item()
            elseif require("luasnip").expand_or_jumpable() then
              require("luasnip").expand_or_jump()
            else
              fallback()
            end
          end, { "i", "s" }),
        },
        formatting = {
          fields = { "menu", "abbr", "kind" },

          format = function(entry, item)
            local entryItem = entry:get_completion_item()
            local color = entryItem.documentation

            if color and type(color) == "string" and color:match "^#%x%x%x%x%x%x$" then
              local hl = "hex-" .. color:sub(2)

              if #vim.api.nvim_get_hl(0, { name = hl }) == 0 then
                vim.api.nvim_set_hl(0, hl, { fg = color })
              end

              item.menu = " "
              item.menu_hl_group = hl
            end

            return item
          end,
        },
      }
      return vim.tbl_deep_extend("force", calculated_opts, require "nvchad.cmp")
    end,
  },
```

**Reply by siduck** [reply]
@Rick-Phoenix 

![image](https://github.com/user-attachments/assets/850064b7-e622-47c8-bb27-12f9a444e047)


please enable tailwind option

**Comment by Rick-Phoenix** [comment]
What do you mean? I can't find any info in the docs on where to enable it...
The lsp is working correctly and in fact I can see the color previews on the class declaration:

![image](https://github.com/user-attachments/assets/c6da1f29-61b2-4603-ad7b-3570d2b8336c)

But in the cmp they don't show up

**Reply by siduck** [reply]
can you show your chadrc? ive enabled it by default. please do lazy sync now

**Reply by Rick-Phoenix** [reply]
Doing lazy sync actually fixed it! Which is strange because this is a fresh install so everything should have been updated... Thanks!
How do I put the color icon on the left? I have set icons_left to true but they are still on the right. 
This is the rest of my chadrc config:

```

---@type ChadrcConfig
local M = {}

M.base46 = {
  theme = "catppuccin",

}

M.ui = {
  cmp = {
    icons_left = true, 
    style = "atom", 
    abbr_maxwidth = 60,
    format_colors = {
      tailwind = true, 
      icon = "󱓻",
    },
},

M.nvdash = {
  load_on_startup = true,
}

return M
```

**Reply by siduck** [reply]
@Rick-Phoenix remove atom. icon_left is just for non atom styles and remove other default config unless u change it. cuz at the end your config gets merged with defautls

**Reply by Rick-Phoenix** [reply]
Aah yeah I tried atom just for a sec before I pasted that to you, changed it now and it works! Thanks :)

**Reply by siduck** [reply]
@Rick-Phoenix if you know about blink.cmp , then you can use that too :p  

{ import = "nvchad.blink.lazyspec" }

( do lazy sync 2x times before this )

**Reply by Rick-Phoenix** [reply]
That looks quite interesting, is it supposed to replace cmp and the nvim-tree entirely or to be used alongside them? The file tree looks very nice

**Reply by siduck** [reply]
@Rick-Phoenix  no only cmp

**Reply by Rick-Phoenix** [reply]
I've tried it for a big but it was really benefit and I couldn't see any benefit compared to nvim-cmp to be honest :/

**Comment by Rick-Phoenix** [comment]
Ok I have redone the configuration for blink properly and it seems to be working well now! It's really fast, and the theme for it looks great!
